saleor/graphql/product/types/products.py

```diff
@@ -699,8 +724,24 @@ def resolve_product_attributes(root: models.ProductType, info):
         return ProductAttributesByProductTypeIdLoader(info.context).load(root.pk)
 
     @staticmethod
-    def resolve_variant_attributes(root: models.ProductType, info):
-        return VariantAttributesByProductTypeIdLoader(info.context).load(root.pk)
+    def resolve_variant_attributes(
+        root: models.ProductType, info, variant_selection: bool = False,
+    ):
+        def apply_variant_selection_filter(attributes):
+            if not variant_selection:
+                return attributes
+            return [
+                attribute
+                for attribute in attributes
+                if attribute.input_type == AttributeInputType.DROPDOWN
+                and attribute.type == AttributeType.PRODUCT_TYPE
+            ]
```

Comment:
I think it would be useful to have a single function that would handle this logic. It could accept a list/qs of `Attribute` instances.

Reply:
We need to reuse this logic in two resolvers, also, in the future we may extend it so it's better to have it in a function.

Reply:
The logic in those 2 resolvers is different, as the data structure is different. I can create a method from part which is used in `PageType`.

Reply:
I introduced a function for this logic.

Reply:
Looks good now. Maybe just add a docstring to that function and it will be perfect :)